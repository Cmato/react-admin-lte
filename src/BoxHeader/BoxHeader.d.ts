import * as React from 'react';

export interface BoxHeaderProps {
  border?: boolean;
  title?: ((props: { className: string; }) => React.ReactNode) | React.ReactNode;
}

declare const BoxHeader: React.ComponentType<BoxHeaderProps>;

export default BoxHeader;
