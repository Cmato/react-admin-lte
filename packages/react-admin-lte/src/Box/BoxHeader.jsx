import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

import BoxUserBlock from './BoxUserBlock';

const propTypes = {
  border: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node,
};

const defaultProps = {
  border: false,
};

const BoxHeader = ({
  className,
  children,
  border
}) => {
  const classes = {
    'box-header': true,
    'with-border': border,
  };

  return (
    <div className={classNames(className, classes)}>
      {children}
    </div>
  );
};

BoxHeader.propTypes = propTypes;
BoxHeader.defaultProps = defaultProps;

BoxHeader.UserBlock = BoxUserBlock;

export default BoxHeader;
