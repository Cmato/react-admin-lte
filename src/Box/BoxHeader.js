import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxHeader = ({
  border,
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    'box-header': true,
    'with-border': border
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

BoxHeader.propTypes = {
  /**
   * If `true`, header will have a bottom border.
   */
  border: PropTypes.bool,
  /**
   * The contents of the box header.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string
};

BoxHeader.defaultProps = {
  border: false
};

export default BoxHeader;
