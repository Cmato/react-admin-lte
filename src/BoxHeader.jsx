import React from 'react';
import classNames from 'classnames';

import BoxTools from './BoxTools';

const propTypes = {
  title: React.PropTypes.string.isRequired,
  border: React.PropTypes.bool,
};

const defaultProps = {
  border: false,
};

const BoxHeader = ({
  className,
  children,
  title,
  border,
}) => {
  const classes = {
    'box-header': true,
    'with-border': border,
  };

  return (
    <div className={classNames(className, classes)}>
      <h3 className="box-title">{title}</h3>
      {children}
    </div>
  );
};

BoxHeader.propTypes = propTypes;
BoxHeader.defaultProps = defaultProps;

BoxHeader.Tools = BoxTools;

export default BoxHeader;
