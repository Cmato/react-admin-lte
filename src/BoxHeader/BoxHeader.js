import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxHeader = ({
  border,
  children,
  className: classNameProp,
  tools: toolsProp,
  ...other
}) => {
  const className = classNames({
    'box-header': true,
    'with-border': border
  }, classNameProp);

  const tools = toolsProp
    ? typeof toolsProp === 'function'
      ? toolsProp({ className: 'box-tools pull-right' })
      : <div className="box-tools pull-right">{toolsProp}</div>
    : null;

  return (
    <div className={className} {...other}>
      {children}
      {tools}
    </div>
  );
};

BoxHeader.propTypes = {
  /**
   * If `true`, header will have a bottom border.
   */
  border: PropTypes.bool,
  /**
   * The contents of the component.
   */
  children: PropTypes.node,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * Box tools to be placed on right side of header. To customize pass a
   * function that will be used to render the tools container.
   */
  tools: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
};

BoxHeader.defaultProps = {
  border: false
};

export default BoxHeader;
