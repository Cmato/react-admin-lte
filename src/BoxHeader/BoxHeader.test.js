import React from 'react';
import classNames from 'classnames';
import { shallow } from 'enzyme';
import BoxHeader from './BoxHeader';

it('renders default', () => {
  expect(shallow(<BoxHeader />))
    .toMatchSnapshot();
});

it('renders title node', () => {
  expect(shallow(<BoxHeader title="I'm a box!" />))
    .toMatchSnapshot();
});

it('renders title function', () => {
  expect(shallow((
    <BoxHeader
      title={({ className: classNameProp }) => {
        const className = classNames(classNameProp, {
          'my-class': true
        });
        return <h2 className={className}>I'm a box!</h2>
      }}
    />
  ))).toMatchSnapshot();
});

it('renders box tools', () => {
  expect(shallow(<BoxHeader><div>I'm a tool!</div></BoxHeader>))
    .toMatchSnapshot();
});
