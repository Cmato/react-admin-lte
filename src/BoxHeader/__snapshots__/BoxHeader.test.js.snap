// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom component 1`] = `
<div
  className="box-header with-border"
>
  <div />
</div>
`;

exports[`renders default 1`] = `
<div
  className="box-header"
>
  <div />
</div>
`;
